delivery_auto_refresh/models/sale_order.py

```diff
@@ -1,21 +1,49 @@
 # Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Carlos Roca
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import api, models
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
 from odoo.tools import safe_eval
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    carrier_id = fields.Many2one(
+        compute="_compute_carrier_id", store=True, readonly=False
+    )
+    invoice_shipping_on_delivery = fields.Boolean(
```

Comment:
I'm not sure we should recover this behavior, but be based on the carrier field that specify the invoicing policy. What do you think?

Reply:
Ok I will use it